src/contrib/mps/gemm.mm

```diff
@@ -0,0 +1,95 @@
+#include "../../runtime/metal/metal_common.h"
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#include <dmlc/logging.h>
+#include <tvm/runtime/device_api.h>
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/util.h>
+
+namespace tvm {
+namespace contrib {
+
+using namespace runtime;
+
+TVM_REGISTER_GLOBAL("tvm.contrib.mps.matmul")
+    .set_body([](TVMArgs args, TVMRetValue *ret) {
+      DLTensor *A = args[0];
+      DLTensor *B = args[1];
+      DLTensor *C = args[2];
+      bool transa = args[3];
+      bool transb = args[4];
+      // call gemm for simple compact code.
+      CHECK_EQ(A->ndim, 2);
+      CHECK_EQ(B->ndim, 2);
+      CHECK_EQ(C->ndim, 2);
+      CHECK(C->strides == nullptr);
+      CHECK(B->strides == nullptr);
+      CHECK(A->strides == nullptr);
+      CHECK(TypeMatch(A->dtype, kDLFloat, 32));
+      CHECK(TypeMatch(B->dtype, kDLFloat, 32));
+      CHECK(TypeMatch(C->dtype, kDLFloat, 32));
+      // Get Metal device API
+      auto func = runtime::Registry::Get("device_api.metal");
+      void *dev_handle = (*func)();
```

Comment:
device API getter need to put into a separate function, with the device handle as static variable.

Reply:
I'm following cudnn api and putting it in a thread-local entry.